File: Seccion 19 - Herencia y Polimorfismo en C++/Seccion_19_Python/01_ejercicio_01.py

class Mamifero:
    
    def __init__(self, anio_nacimiento, lugar_nacimiento, raza):
        self.anio_nacimiento = anio_nacimiento
        self.lugar_nacimiento = lugar_nacimiento
        self.raza = raza
    
    def imprimir_dieta(self):
        pass
    
    def get_anio_nacimiento(self):
        return self.anio_nacimiento

    def get_lugar_nacimiento(self):
        return self.lugar_nacimiento
    
    def get_raza(self):
        return self.raza

class Felino(Mamifero):
    
    def __init__(self, anio_nacimiento, lugar_nacimiento, raza, nombre_circo):
        super().__init__(anio_nacimiento, lugar_nacimiento, raza)
        self.nombre_circo = nombre_circo
    
    def imprimir_dieta(self):
        return "La dieta del felino es a base de carne.\n"

    def set_nombre_circo(self, nuevo_nombre):
        self.nombre_circo = nuevo_nombre
    
    def get_nombre_circo(self):
        return self.nombre_circo

class GatoDomestico(Felino):
    
    def __init__(self, anio_nacimiento, lugar_nacimiento, raza, nombre_duenio):
        super().__init__(anio_nacimiento, lugar_nacimiento, raza, "")
        self.nombre_duenio = nombre_duenio
    
    def imprimir_dieta(self):
        return "La dieta del gato es con Whiskas\n"

    def set_nombre_duenio(self, nuevo_nombre):
        self.nombre_duenio = nuevo_nombre
    
    def get_nombre_duenio(self):
        return self.nombre_duenio

def main():
    
    minino = GatoDomestico(2015, "Trujillo", "Gato Persa", "Diego")
    estrella_circo = Felino(2014, "La India", "Tigre de Bengala", "Espectacular")
    
    print("\nDietas:")
    print(f"Gato: {minino.imprimir_dieta()}Felino: {estrella_circo.imprimir_dieta()}")
    
    print("\nAños de nacimiento y lugar de nacimiento: ")
    print(f"Gato: {minino.get_anio_nacimiento()} - {minino.get_lugar_nacimiento()}")
    print(f"Felino: {estrella_circo.get_anio_nacimiento()} - {estrella_circo.get_lugar_nacimiento()}")
    
    print("\nCambiar el nombre del dueño del gato:")
    nuevo_nombre = input("Digite el nombre del nuevo dueño: ")
    minino.set_nombre_duenio(nuevo_nombre)
    print(f"El nuevo dueño del Gato es: {minino.get_nombre_duenio()}")
    
    print("\nRaza:")
    print(f"Gato: {minino.get_raza()}\nFelino: {estrella_circo.get_raza()}")
    
    print("\nCambiar el nombre del Circo:")
    nuevo_nombre = input("Digite el nuevo nombre del circo: ")
    estrella_circo.set_nombre_circo(nuevo_nombre)
    print(f"El nuevo nombre del circo es: {estrella_circo.get_nombre_circo()}")

if __name__ == "__main__":
    main()

File: Seccion 19 - Herencia y Polimorfismo en C++/Seccion_19_Dart/01_ejercicio_01.dart

import 'dart:io';

abstract class Mamifero {

  int _anioNacimiento;
  String _lugarNacimiento, _raza;

  Mamifero(this._anioNacimiento, this._lugarNacimiento, this._raza);

  String imprimirDieta();

  int getAnioNacimiento() {
    return _anioNacimiento;
  }

  String getLugarNacimiento() {
    return _lugarNacimiento;
  }

  String getRaza() {
    return _raza;
  }
}

class Felino extends Mamifero {

  String nombreCirco;

  Felino(int anioNacimiento, String lugarNacimiento, String raza, this.nombreCirco) : super(anioNacimiento, lugarNacimiento, raza);

  @override
  String imprimirDieta() {
    return "La dieta del felino es a base de carne.\n";
  }

  void setNombreCirco(String nuevoNombre) {
    nombreCirco = nuevoNombre;
  }

  String getNombreCirco() {
    return nombreCirco;
  }

}

class GatoDomestico extends Felino {

  String nombreDuenio;

  GatoDomestico(int anioNacimiento, String lugarNacimiento, String raza, this.nombreDuenio) : super(anioNacimiento, lugarNacimiento, raza, "");

  @override
  String imprimirDieta() {
    return "La dieta del gato es con Whiskas.\n";
  }

  void setNombreDuenio(String nuevoNombre) {
    nombreDuenio = nuevoNombre;
  }

  String getNombreDuenio() {
    return nombreDuenio;
  }
}

void main() {

  final GatoDomestico minino = new GatoDomestico(2015, "Trujillo", "Gato Persa", "Diego");
  final Felino estrellaCirco = new Felino(2014, "La India", "Tigre de Bengala", "Espectacular");
    
  print("\nDietas:");
  print("Gato: ${minino.imprimirDieta()}Felino: ${estrellaCirco.imprimirDieta()}");
    
  print("\nAños de nacimiento y lugar de nacimiento: ");
  print("Gato: ${minino.getAnioNacimiento()} - ${minino.getLugarNacimiento()}");
  print("Felino: ${estrellaCirco.getAnioNacimiento()} - ${estrellaCirco.getLugarNacimiento()}");
    
  print("\nCambiar el nombre del dueño del gato:");
  String nuevoNombre;
  stdout.write("Digite el nombre del nuevo dueño: ");
  nuevoNombre = stdin.readLineSync();
  minino.setNombreDuenio(nuevoNombre);
  print("El nuevo dueño del Gato es: ${minino.getNombreDuenio()}");
    
  print("\nRaza:");
  print("Gato: ${minino.getRaza()}\nFelino: ${estrellaCirco.getRaza()}");
    
  print("\nCambiar el nombre del Circo:");
  stdout.write("Digite el nuevo nombre del circo: ");
  nuevoNombre = stdin.readLineSync();
  estrellaCirco.setNombreCirco(nuevoNombre);
  print("El nuevo nombre del circo es: ${estrellaCirco.getNombreCirco()}");
}